### PR TITLE
docs: expand the section on commit messages

### PR DESCRIPTION
**Pull request by mvo5**
This comit expand on the commit messages section and makes it more clear why they are important and what should be part of them.

There are many good guides on this so it's fine to keep this brief but I think it was a little bit to concise :)

It also links to a good commit message example, I am open for suggestions here though, while this is good (as pretty much any from Christian) I'm sure there are even better ones.

### CODE REVIEW

**Review by ondrejbudai — DISMISSED** [review]
:heart_eyes: Thanks!

**Review by croissanne — DISMISSED** [review]

File: docs/developer-guide/01-general/workflow.md [hunk 1]
```diff
@@ -11,12 +11,23 @@
...
 should summarize what the commit is doing.
 - GitHub truncates the first line if it's longer than 65 characters, which is something to keep in mind as well.
+- GitHub PR descriptions are disconnected from the actual `git`
+history and will not show up in e.g. `git blame` or `git log` so consider
+them more ephemeral.
+
+A good commit message will help our future selfs to get back into the
+context that the author had when writing the code. It should include
+decisions/considerations/background that were important when the
+commit was writen so that a `git show <commit>` provides a concise
```
Comment (schuellerf) [inline docs/developer-guide/01-general/workflow.md]:
```suggestion
commit was written so that a `git show <commit>` provides a concise
```

**Review by schuellerf — CHANGES_REQUESTED** [review]
I like it, thanks!
…just a minor typo

File: docs/developer-guide/01-general/workflow.md (hunk 1 above)
Comment (mvo5) [reply docs/developer-guide/01-general/workflow.md]: Thanks! Pushed an update

**Review by schuellerf — DISMISSED** [review]
lgtm, thanks!

File: docs/developer-guide/01-general/workflow.md [hunk 2]
```diff
@@ -11,12 +11,23 @@
...
 ```
 
 - The `<module>` should point to the area of the codebase (for instance `tests` or `tools`). The topic
 should summarize what the commit is doing.
 - GitHub truncates the first line if it's longer than 65 characters, which is something to keep in mind as well.
+- GitHub PR descriptions are disconnected from the actual `git`
+history and will not show up in e.g. `git blame` or `git log` so consider
+them more ephemeral.
+
+A good commit message will help our future selfs to get back into the
```
Comment (achilleas-k) [inline docs/developer-guide/01-general/workflow.md]:
```suggestion
A good commit message will help our future selves to get back into the
```

File: docs/developer-guide/01-general/workflow.md [hunk 3]
```diff
@@ -11,12 +11,23 @@
...
+context that the author had when writing the code. It should include
+decisions/considerations/background that were important when the
```
Comment (achilleas-k) [inline docs/developer-guide/01-general/workflow.md]:
```suggestion
decisions, considerations, and background information that were important when the
```

File: docs/developer-guide/01-general/workflow.md [hunk 4]
```diff
@@ -11,12 +11,23 @@
...
+commit was written so that a `git show <commit>` provides a concise
+way to (re)acquire this context. See [this](https://github.com/osbuild/osbuild/commit/a2e212bb2641cf28e5701ad4a2202261c2c5ee5c) as an example.
```
Comment (achilleas-k) [inline docs/developer-guide/01-general/workflow.md]:
```suggestion
way to (re)acquire this context. See [this commit message](https://github.com/osbuild/osbuild/commit/a2e212bb2641cf28e5701ad4a2202261c2c5ee5c) as an example.
```

**Review by achilleas-k — COMMENTED** [review]
Some suggestions for potential improvements, if you agree.

File: docs/developer-guide/01-general/workflow.md (hunk 4 above)
Comment (achilleas-k) [reply docs/developer-guide/01-general/workflow.md]: Or
```suggestion
way to (re)acquire this context. See [this commit](https://github.com/osbuild/osbuild/commit/a2e212bb2641cf28e5701ad4a2202261c2c5ee5c) as an example.
```

**Review by ochosi — DISMISSED** [review]
@achilleas-k Already left some nice wording suggestions - I'm all good with the content!

**Review by achilleas-k — APPROVED** [review]
Wonderful!!

**Pull request merged**